lib/usecases/environment/GetAllEnvironmentsUseCase.js

```diff
@@ -56,14 +88,92 @@ class GetAllEnvironmentsUseCase {
+                    query: ENVIRONMENT_LATEST_HISTORY_ITEM_SUBQUERY,
+                    alias: 'currentStatus',
+                });
+            }
+
+            if (statusHistory) {
+                // Split the string into separate characters
+                const filters = statusHistory.split('');
+
+                // Swap the acronyms with the status (=acronym -> status)
+                const acronymToStatus = {};
+                for (const status in statusAcronyms) {
+                    acronymToStatus[statusAcronyms[status]] = status;
+                }
+
+                // Create a list of status history with the status acronyms
+                const statusFilters = [];
+                for (let i = 0; i < filters.length; i++) {
+                    // Only add the next status if it is in the acronyms list
+                    if (Object.values(statusAcronyms).includes(filters[i])) {
+                        // Use the acronym to get the status
+                        statusFilters.push(acronymToStatus[filters[i]]);
+                    }
+                }
+
+                // Filter the environments by status history using the subquery
+                queryBuilder.literalWhere(
+                    `${ENVIRONMENT_STATUS_HISTORY_SUBQUERY} = :statusFilters`,
+                    // Create a string of the status filters separated by a comma
+                    { statusFilters: statusFilters.join(',') },
+                );
+
+                queryBuilder.includeAttribute({
+                    query: ENVIRONMENT_STATUS_HISTORY_SUBQUERY,
+                    alias: 'statusHistory',
+                });
+            }
+
+            if (runNumbers) {
+                // Convert the string of run numbers to an array of numbers
+                const filters = runNumbers.split(',').map((filter) => Number(filter.trim()));
+
+                // Filter should be like with only one filter
+                if (filters.length === 1) {
+                    // Filter the environments by the run numbers with the association
+                    queryBuilder.include({
+                        association: 'runs',
+                        // Select only the environments that have a run with the given run number
+                        where: {
+                            runNumber: { [Op.substring]: filters },
+                        },
+                    });
+                }
+
+                // Filters should be exact with more than one filter
+                if (filters.length > 1) {
+                    // Filter the environments by the run numbers with the association
+                    queryBuilder.include({
+                        association: 'runs',
+                        // Select only the environments that have a run with the given run number
+                        where: {
+                            runNumber: { [Op.in]: filters },
+                        },
+                    });
+                }
```

Comment:
```suggestion
                // Filter should be like with only one filter
                if (filters.length) {
                    queryBuilder.include({
                        association: 'runs',
                        where: {
                            runNumber: { [length === 1 ? Op.substring : Op.in]: filters },
                        },
                    });
                }
```

Reply:
implemented, thank you for the suggestion!